pydicom/uid.py

```diff
@@ -168,26 +188,28 @@ def generate_uid(prefix=pydicom_root_UID, truncate=False):
     <http://dicom.offis.de/dcmtk.php.en>`_.
 
     :param prefix: The site root UID. Default to pydicom root UID.
+    :param entropy_srcs: A list of one of more strings that are hashed to
+    generate the suffix
     '''
     max_uid_len = 64
 
     if prefix is None:
-        dicom_uid = '2.25.{0}'.format(uuid.uuid1().int)
+        prefix = '2.25.'
     else:
-        uid_info = [uuid.getnode(),
-                    fabs(os.getpid()),
-                    datetime.datetime.today().second,
-                    datetime.datetime.today().microsecond]  # nopep8
-
-        suffix = ''.join([str(int(x)) for x in uid_info])
-        dicom_uid = ''.join([prefix, suffix])
-
-    if truncate:
-        dicom_uid = dicom_uid[:max_uid_len]
-
-    dicom_uid = UID(dicom_uid)
-
-    # This will raise an exception if the UID is invalid
-    dicom_uid.is_valid()
-
-    return dicom_uid
+        if len(prefix) > max_uid_len - 1:
+            raise ValueError("The prefix must be less than 63 chars")
```

Comment:
I was considering putting a warning here if the prefix is too long since that could make collisions likely, but I guess it isn't clear where to draw the line.